mava/wrappers/jaxmarl.py

```diff
@@ -252,6 +253,12 @@ def _create_observation(
         return Observation(**obs_data)
 
     def observation_spec(self) -> specs.Spec:
+        if isinstance(self._env, SimpleSpreadMPE):
+            obs, _ = self._env.reset(jax.random.PRNGKey(0))
+            # The shape provided in the `observation_spaces` isn't always the same as
+            # that given after constructing the full agent's obs.
+            self._env.observation_spaces["agent_0"].shape = obs["agent_0"].shape
```

Comment:
I think this is fixed in jaxmarl 0.0.5, if that is the case could we please update and remove this :pray:

Reply:
Yes this is fixed in that version but we should handle other issues related to respecting a specific version of Jax, Flax and else a dependencies conflict and this error: https://github.com/FLAIROx/JaxMARL/issues/108.
It works for me if jax==0.4.25 and flax should be forced to 0.8 since the one forced by jaxmarl 0.0.5 gives the same previous issue.

Reply:
This really sucks. Ye I think they just haven't upgraded flax for their algorithms, which seems to be why they pin to 0.4.25, I'm not really sure what to do here, because I'd really like to pin to the latest jaxmarl and definitely latest jax. @RuanJohn thoughts?

Reply:
Since we are using the forked version of JaxMARL now, do we still want to update this?

Reply:
Yesss I merged the new changes already 🚀